src/network/networkd-link.c

```diff
@@ -1357,7 +1357,7 @@ static int link_set_flags(Link *link) {
 
         if (link->network->arp >= 0) {
                 ifi_change |= IFF_NOARP;
-                ifi_flags |= IFF_NOARP;
+                ifi_flags |= link->network->arp ? IFF_NOARP : 0;
```

Comment:
wrong way round. if arp is > 0 we should set the flags to 0, and if it is == 0 we should set IFF_NOARP...